commands/trade.js

```diff
@@ -467,6 +466,10 @@ module.exports = function container (get, set, clear) {
         function forwardScan () {
           function saveSession () {
             engine.syncBalance(function (err) {
+              if (!err && s.balance.asset === undefined) {
```

Comment:
My worry is that some exchanges might return undefined for empty balances, and we would tell the user their API key is not working even though it is. My worry is just an assumption though so if nobody shares this worry I'm fine with merging this.

Reply:
I thought so as well, but if asset or currency is not defined in that balance object, then a lot of other code will crash for sure. This way or that way it dies. :(